soccer/planning/Path.cpp

```diff
@@ -32,4 +37,42 @@ void Path::draw(SystemState* const state, const QColor& color,
     addPoint(end().motion);
 }
 
-}  // namespace Planning
+std::unique_ptr<ConstPathIterator> Path::iterator(RJ::Time startTime,
+                                                  float deltaT) const {
+    return std::move(
+        std::make_unique<ConstPathIterator>(this, startTime, deltaT));
+}
+
+bool Path::pathsIntersect(const std::vector<const Path*>& paths, float* hitTime,
+                          Geometry2d::Point* hitLocation,
+                          RJ::Time startTime) const {
+    const float deltaT = 0.05;
+    const float hitRadius = Robot_Radius * 2.5f;
```

Comment:
Well we use it [here](https://github.com/RoboJackets/robocup-software/blob/master/soccer/gameplay/plays/offense/basic_122.py#L167) and [here](https://github.com/RoboJackets/robocup-software/blob/master/soccer/gameplay/plays/restarts/their_restart.py#L91) currently.

It may be ok to not have per-robot avod radii, but if we decide to stop using the self avoid mask for anything, we should get rid of it and the related methods.